src/ring_theory/localization.lean

```diff
@@ -602,23 +618,128 @@ mul_equiv_of_quotient_symm_monoid_of _
 end localization
 variables {M}
 
+section at_prime
+
+variables (I : ideal R) [hp : I.is_prime]
+include hp
+namespace ideal
+
+/-- The complement of a prime ideal `I ⊆ R` is a submonoid of `R`. -/
+def prime_submonoid :
```

Comment:
```suggestion
def prime_complement :
```
Otherwise it sounds like you're turning the prime ideal itself into a submonoid.

Reply:
good point... What about ```prime_compl```, to be in keeping with ```set.compl``` etc?

Reply:
yup, sounds good to me